norduser/group_monitor.go

```diff
@@ -106,82 +122,142 @@ func getUID(username string) (userIDs, error) {
 	}, nil
 }
 
-func (n *NordvpnGroupMonitor) handleGroupUpdate(currentGroupMembers userSet, newGroupMembers userSet) {
-	for member := range currentGroupMembers {
-		// member not modified, do nothing
-		if _, ok := newGroupMembers[member]; ok {
-			newGroupMembers[member] = true
+func parseSessions(sessions string) []string {
+	re := regexp.MustCompile(`(?m)^(\S+)`)
+	matches := re.FindAllStringSubmatch(sessions, -1)
+	if matches == nil {
+		return []string{}
+	}
+
+	activeUsers := []string{}
+	for _, match := range matches {
+		activeUsers = append(activeUsers, match[1])
+	}
+
+	return activeUsers
+}
+
+func updateGroupMembersState(groupMembers userSet, activeUsers []string) userSet {
+	for member := range groupMembers {
+		if idx := slices.Index(activeUsers, member); idx != -1 {
+			groupMembers[member] = userActive
+		}
+	}
+
+	return groupMembers
+}
+
+// NordVPNGroupMonitor monitors the nordvpn system group and starts/stops norduserd for users added/removed from the
+// group.
+type NordVPNGroupMonitor struct {
+	norduserd service.NorduserService
+	isSnap    bool
+	userIDGetter
+}
+
+func NewNordVPNGroupMonitor(service service.NorduserService) NordVPNGroupMonitor {
+	return NordVPNGroupMonitor{
+		norduserd:    service,
+		isSnap:       snapconf.IsUnderSnap(),
+		userIDGetter: osGetter{},
+	}
+}
+
+func (n *NordVPNGroupMonitor) handleGroupUpdate(currentGroupMembers userSet, newGroupMembers userSet) userSet {
+	for member, state := range currentGroupMembers {
+		// member is still in group and session is active, do nothing
+		if newState, ok := newGroupMembers[member]; ok && newState == userActive {
```

Comment:
there might be some issues here:
* connect with SSH to the computer, before logging in from the graphical interface
* while connected login to the desktop
In this case the tray doesn't appear, because the process is already started for ssh session. Also disconnecting from the ssh would still not display the tray.